### PR TITLE
SOLR-15324 Upgrade Jaeger dependency from 1.1.0 to 1.6.0

### PR DESCRIPTION
**Pull request by janhoy**
This also upgrades libthrift to 0.14.1, fixing CVE-2020-13949

https://issues.apache.org/jira/browse/SOLR-15324

### CODE REVIEW

**Comment by janhoy** [comment]
Test seem to pass, but I don't know if the upgrade will actually work with a live Jaeger server. Anyone who have the chance to test?

File: versions.lock [hunk 1]
```diff
@@ -174,11 +174,13 @@ org.apache.poi:poi:4.1.1 (1 constraints: 08050336)
...
 org.apache.poi:poi-ooxml-schemas:4.1.1 (1 constraints: 08050336)
 org.apache.poi:poi-scratchpad:4.1.1 (1 constraints: 08050336)
-org.apache.thrift:libthrift:0.13.0 (1 constraints: 8e0dfd33)
+org.apache.thrift:libthrift:0.14.1 (1 constraints: 900d0134)
 org.apache.tika:tika-core:1.24 (1 constraints: db04f730)
 org.apache.tika:tika-java7:1.24 (1 constraints: db04f730)
 org.apache.tika:tika-parsers:1.24 (1 constraints: db04f730)
 org.apache.tika:tika-xmp:1.24 (1 constraints: db04f730)
+org.apache.tomcat:tomcat-annotations-api:8.5.46 (1 constraints: 5a1162ea)
+org.apache.tomcat.embed:tomcat-embed-core:8.5.46 (1 constraints: 780c4b05)
```
Comment (janhoy) [inline versions.lock:183]: Would it be safe to exclude these two new tomcat dependencies or are they needed by our use of jaeger?

Comment (janhoy) [reply versions.lock]: I excluded them and tests still pass..

**Review by dsmiley — APPROVED** [review]
Thanks for minding the exclusions.

**Pull request merged**